packages/state/recoil/selectors/wallet.ts

```diff
@@ -50,3 +54,73 @@ export const walletCw20BalancesSelector = selectorFamily<
       }))
     },
 })
+
+export const walletTokenStakedDenomsSelector = selectorFamily<
+  readonly string[],
+  WithChainId<{ walletAddress: string }>
+>({
+  key: 'walletTokenStakedDenoms',
+  get:
+    ({ walletAddress, chainId }) =>
+    ({ get }) => {
+      // Get the DAO's that the wallet is a member of
+      const daos = get(
+        queryWalletIndexerSelector({
+          chainId,
+          walletAddress,
+          formula: 'daos/memberOf',
+        })
+      )
+      if (!daos || !Array.isArray(daos) || daos.length === 0) {
+        return []
+      }
+
+      // Get the token staked voting modules for each DAO
+      const votingModules = get(
+        waitForAll(
+          daos.map(({ dao: contractAddress }) =>
+            votingModuleSelector({
+              contractAddress,
+              chainId,
+              params: [],
+            })
+          )
+        )
+      ).filter((contractAddress) =>
+        get(
+          isContractSelector({
+            contractAddress,
+            chainId,
+            names: DAO_VOTING_TOKEN_STAKED_CONTRACT_NAMES,
+          })
+        )
+      )
+
+      if (
+        !votingModules ||
+        !Array.isArray(votingModules) ||
+        votingModules.length === 0
+      ) {
```

Comment:
```suggestion
      if (votingModules.length === 0) {
```

no need for these conditions as `votingModules` is always defined and always an array.

the reason we need to check `daos` above in this way is because it's returned from an external API (our indexer), and we want to safeguard against any unexpected responses.